pkg/ipcache/metadata.go

```diff
@@ -544,7 +537,16 @@ func (ipc *IPCache) UpdatePolicyMaps(ctx context.Context, addedIdentities, delet
 func (ipc *IPCache) resolveIdentity(ctx context.Context, prefix netip.Prefix, info prefixInfo, restoredIdentity identity.NumericIdentity) (*identity.Identity, bool, error) {
 	// Override identities always take precedence
 	if identityOverrideLabels, ok := info.identityOverride(); ok {
-		return ipc.IdentityAllocator.AllocateIdentity(ctx, identityOverrideLabels, false, identity.InvalidIdentity)
+		id, isNew, err := ipc.IdentityAllocator.AllocateIdentity(ctx, identityOverrideLabels, false, identity.InvalidIdentity)
+		if err != nil {
+			log.WithError(err).WithFields(logrus.Fields{
+				logfields.IPAddr: prefix,
+				logfields.Labels: identityOverrideLabels,
+			}).Warning(
+				"Failed to allocate new identity while handling change in labels associated with a prefix.",
```

Comment:
I'd suggest making the log msgs unique so that we can locate where errors are coming from